include/lsst/daf/base/DateTime.h

```diff
@@ -57,44 +61,230 @@ namespace base {
 
 class DateTime {
 public:
-    enum Timescale { TAI, UTC, TT };
-    enum DateSystem { JD, MJD, EPOCH };
-    explicit DateTime(long long nsecs = 0LL, Timescale scale = TAI);
-    explicit DateTime(double date, DateSystem system = MJD, Timescale scale = TAI);
-    DateTime(int year, int month, int day, int hr, int min, int sec,
-             Timescale scale = TAI);
+    enum DateSystem { JD=0, MJD, EPOCH };
+    enum Timescale { TAI=5, UTC, TT };  // use values that do not overlap DateSystem
+                                        // to avoid confusing one for the other in Python
+    // an invalid DateTime has _nsec == invalid_nsecs
+    constexpr static long long invalid_nsecs = std::numeric_limits<std::int64_t>::min();
+
+    /**
+     * Default constructor: construct an invalid DateTime
+     */
+    explicit DateTime();
+
+    /**
+     * Construct a DateTime from nanoseconds since the unix epoch
+     *
+     * @param[in] nsecs  integer nanoseconds since the unix epoch;
+                         if nsecs == DateTime.invalid_nsecs then the DateTime is invalid,
+                         regardless of scale
+     * @param[in] scale  time scale of input (TAI, TT or UTC, default TAI).
+     * @throw lsst.pex.exceptions.DomainError if scale is UTC and the date is before 1961-01-01
+     */
+    explicit DateTime(long long nsecs, Timescale scale=TAI);
+
+    /**
+     * Construct a DateTime from a double in the specified system and scale
+     *
+     * @param[in] date  specified date
+     * @param[in] system  time system of input (JD, MJD, or EPOCH)
+     * @param[in] scale  time scale of input (TAI, TT or UTC, default TAI).
+     * @throw lsst.pex.exceptions.DomainError if scale is UTC and the date is before 1961-01-01
+     */
+    explicit DateTime(double date, DateSystem system=MJD, Timescale scale=TAI);
+
+    /**
+     * Construct a DateTime from year, month, day, etc. (tm struct fields)
+     *
+     * @param[in] year   year; must be in the range [1902, 2261], inclusive.
+     * @param[in] month  month number, where January = 1
+     * @param[in] day  day of the month (1 to 31).
+     * @param[in] hr  hour (0 to 23)
+     * @param[in] min  minute (0 to 59)
+     * @param[in] sec  integer seconds (0 to 60)
+     * @param[in] scale  time scale of input (TAI, TT or UTC, default TAI).
+     * @throw lsst.pex.exceptions.DomainError if scale is UTC and the date is before 1961-01-01,
+     *      or the year < 1902 or > 2261
```

Comment:
This reads somewhat confusingly as if the UTC statement also applies to the 1902 and 2261. Perhaps move them forward?